bokehjs/test/test_common/test_ui_events.coffee

```diff
@@ -0,0 +1,63 @@
+{expect} = require "chai"
+utils = require "../utils"
+jsdom = require 'mocha-jsdom'
+cheerio = require 'cheerio'
+sinon = require 'sinon'
+
+ui_events = utils.require "common/ui_events"
+# Stub out _hammer_element as not used in testing
+sinon.stub(ui_events.prototype, "_hammer_element") 
+
+
+describe "ui_events", ->
+  jsdom()
+
+  html = '<body><canvas></canvas></body>'
+  $ = cheerio.load html
+
+  beforeEach ->
+    e = new Event "wheel",
+
+    deltaY: 100
+    deltaX: 100
+    e.bokeh = {}
+
+    @preventDefault = sinon.spy(e, "preventDefault")
+    @stopPropagation = sinon.spy(e, "stopPropagation")
+    @e = e
+
+    @ui_event = new ui_events({
+      hit_area: $('canvas')
+      tool_manager: gestures: scroll: active: true
```

Comment:
Is this a deeply nested object? If so, would you mind rewriting this more explicitly.